client/ayon_core/hosts/nuke/api/utils.py

```diff
@@ -142,3 +149,160 @@ def is_headless():
         bool: headless
     """
     return QtWidgets.QApplication.instance() is None
+
+
+def create_error_report(context):
+    error_message = ""
+    success = True
+    for result in context.data["results"]:
+        if result["success"]:
+            continue
+
+        success = False
+
+        err = result["error"]
+        formatted_traceback = "".join(
+            traceback.format_exception(
+                type(err),
+                err,
+                err.__traceback__
+            )
+        )
+        fname = result["plugin"].__module__
+        if 'File "<string>", line' in formatted_traceback:
+            _, lineno, func, msg = err.traceback
+            fname = os.path.abspath(fname)
+            formatted_traceback = formatted_traceback.replace(
+                'File "<string>", line',
+                'File "{0}", line'.format(fname)
+            )
+
+        err = result["error"]
+        error_message += "\n"
+        error_message += formatted_traceback
+
+    return success, error_message
+
+
+def submit_headless_farm(node):
+    # Ensure code is executed in root context.
+    if nuke.root() == nuke.thisNode():
+        _submit_headless_farm(node)
+    else:
+        # If not in root context, move to the root context and then execute the
+        # code.
+        with nuke.root():
+            _submit_headless_farm(node)
+
+
+def _submit_headless_farm(node):
+    context = util.collect()
```

Comment:
> I feel that in a way these two features are rather similar and it would be great to unify where we can and streamline the "publishing API" where it appears to be lacking?

Think this will be outside the scope of this PR unfortunately.

Reply:
> This not going through the CreateContext means you might lack instance data that it needs from the new publisher Create Context. I really wonder whether this logic is 'safe' and should be more re-written to at least started of with the Create Context.

@iLLiCiTiT ?

Reply:
Do we really need it to be using **CreateContext** @iLLiCiTiT, this functionality's aim is not publishing. It only needs to simply submit job to farm, with no validation what so ever. This should be crude as much as possible.

Reply:
> It only needs to simply submit job to farm, with no validation what so ever.

Not entirely true. We want validations and go through the publishing code as much as possible, so we do not duplicate code or reinvent some logic.

Reply:
I really have my doubts whether it's worth maintaining this much code for whatever this seems to be doing. I think this code is a clear sign of how "bad" the situation is currently with the publishing system and how much of an easy API lacks for that - as others have commented frequently. I believe @dee-ynput has been made aware of that and @fabiaserra has been vocal about that lack of API as well.

Not saying I have any issues with merging this (we don't use Nuke anyway) - but just saying that this shows the issue at hand. This looks like something that in essence should've been a few lines of code away to implement.

Reply:
> Think this will be outside the scope of this PR unfortunately.

Behind this argument and pressure from clients you guys end up merging so much half-done patchy stuff and then the proper fix is never done... arguing that there's no time or resources to do it right and everytime you do this it gets harder and harder to realign and clean the design, even more so because of the policy you guys have of creating simple small PRs that disencourage doing anything big, like I said here https://github.com/ynput/ayon-shotgrid/pull/83#issuecomment-2045353485

Reply:
Clear example of what I'm saying is this comment I added warning about the changes, https://github.com/ynput/OpenPype/pull/4903#issuecomment-1738343017 it was merged and never done anything else about it

Reply:
Ad not using `CreateContext`. The context is auto-created during collection, so it is not really necessary, but what I'm worried about is the approach how it tells to use farm publishing, and skip some other stuff.

Clearing `"families"` is dangerous, change `"farm"` out of plugins is dangerous, changeing `"transfer"` (to `False` for some reason?) is dangerous. Forcing `"use_published_workfile"` to `False` is strange too.